docs/content/tutorials/create-rest-api.md

```diff
@@ -0,0 +1,299 @@
+---
+title: Creating and Consuming REST API
+category: tutorial
+menu_order: 4
+---
+
+# Creating REST API
+
+Starting from the [how to start guide](how-to-start.html), we will create a Book API to get, insert, update, and delete Books. 
+
+First we need set up the application to handle api routes. Uncomment `forward "/api" apiRouter` line in `appRouter` inside Router.fs. Then, uncomment the `api` pipeline and `apiRouter` in the same file. Finally, change `forward "/someApi" someScopeOrController` to  `forward "/books" Books.Controller.apiRouter` inside `apiRouter`. 
+
+We also need to clean up the template. Remove the `error_handler (text "Api 404")` line inside Router.fs. Remove `open FSharp.Control.Tasks.ContextInsensitive` from BookRepository.fs and BookController.fs file inside the Books folder. Add `open Giraffe` to BookController.fs. Finally, replace `open Giraffe.GiraffeViewEngine` with `open Giraffe.ViewEngine` inside BooksViews.fs.
+
+## Configure Endpoint Routing
+
+We will be using Endpoint Routing for the application. See [here](https://saturnframework.org/explanations/endpoint-routing.html) for more information. To do this. Add `open Saturn.Endpoint` to the Router.fs and BooksController.fs files. Then inside Program.fs, replace `use_router Router.appRouter` with `use_endpoint_router Router.appRouter`.
+
+## API Router
+
+We need to create router for our API endpoints. Create the following `apiRouter` at the bottom inside BooksController.fs to handle the standard GET, POST, PUT, DELETE.
+
+```fsharp
+let apiRouter =
+    router {
+        get "/" getAction
+        getf "/%s" (fun id -> getByIdAction id)
+        post "/" (bindJson<Book> (fun book -> postAction book))
+        putf "/%s" (fun id -> (bindJson<Book> (fun book -> putAction id book)))
+        deletef "/%s" (fun id -> deleteApiAction id)
+    }
+```
+
+With the setup done earlier, this will set the base of our Books API endpoints at `"http://localhost:8085/api/books/"`.
+
+We have not create the action functions yet. This router will automatically pass `HttpFunc` and `HttpContext` to our functions. All our action functions will have those as our parameter.
+
+For GET, we need no id to get all book so the route will just the current location. Because of `forward "/books" Books.Controller.apiRouter` inside our Router.fs file. The path for our GET request is `"http://localhost:8085/api/books/"`

```

Comment:
~~For GET, we need no id to get all book so the route will just the current location. Because of `forward "/books" Books.Controller.apiRouter` inside our Router.fs file. The path for our GET request is `"http://localhost:8085/api/books/"`~~

We don't need an id to return a list of all books so the path for our GET request is `"http://localhost:8085/api/books/"` thanks to `forward "/books" Books.Controller.apiRouter` inside our Router.fs file.